File: UtilitiesVB/Solution Interfaces/IListOfIDs.vb
Imports System.Numerics
Imports Microsoft.Office.Interop.Outlook

Public Interface IListOfIDs
    Property Filepath As String
    ReadOnly Property MaxIDLength As Long
    Property UsedIDList As List(Of String)
    Sub CompressToDoIDs(OlApp As Application)
    Sub RefreshIDList(Application As Application)
    Sub Save()
    Sub Save(Filepath As String)
    Function ConvertToBase(nbase As Integer, num As BigInteger, Optional intMinDigits As Integer = 2) As String
    Function ConvertToDecimal(nbase As Integer, strBase As String) As BigInteger
    Function FlattenArry(varBranch() As Object) As String
    Function GetMaxToDoID() As String
    Function GetNextAvailableToDoID(strSeed As String) As String
End Interface

File: UtilitiesVB/Solution Interfaces/IListOfIDs.cs
using System.Collections.Generic;
using System.Numerics;
using Microsoft.Office.Interop.Outlook;

namespace UtilitiesVB
{

    public interface IListOfIDs
    {
        string Filepath { get; set; }
        long MaxIDLength { get; }
        List<string> UsedIDList { get; set; }
        void CompressToDoIDs(Application OlApp);
        void RefreshIDList(Application Application);
        void Save();
        void Save(string Filepath);
        string ConvertToBase(int nbase, BigInteger num, int intMinDigits = 2);
        BigInteger ConvertToDecimal(int nbase, string strBase);
        string GetMaxToDoID();
        string GetNextAvailableToDoID(string strSeed);
        void SubstituteIdRoot(string oldPrefix, string newPrefix);
    }
}